cmd/kubelet/app/server.go

```diff
@@ -339,6 +227,88 @@ func validateConfig(s *options.KubeletServer) error {
 	return nil
 }
 
+// if the node config controller was used, it will be returned as the first return value
+func initNodeConfig(s *options.KubeletServer, client clientset.Interface, nodeName string, standaloneMode bool) (*nodeconfig.NodeConfigController, error) {
```

Comment:
this function that mutates global flags (possibly twice) and mutates the config held by the server passed in is really difficult to understand. I'd rather keep the feature flag and `s.KubeletConfiguration` changes in the main run function so the flow is linear, and keeps this function side-effect free.